src/cpp/r/R/Api.R

```diff
@@ -841,6 +841,33 @@ options(terminal.manager = list(terminalActivate = .rs.api.terminalActivate,
    .Call("rs_systemUsername", PACKAGE = "(embedding)")
 })
 
+# store callback functions to be executed after a specified chunk
+# and return a handle to unregister the chunk
+.rs.addApiFunction("registerChunkCallback", function(chunkCallback = function(chunkName, chunkCode){}) {
+
+   if (!is.function(chunkCallback))
+      stop("'chunkCallback' must be a function")
+
+   # if one does not already exist, add an environment which will host registered callbacks
+   if (!exists(".rs.notebookChunkCallbacks"))
+      assign(".rs.notebookChunkCallbacks",
+             value = new.env(parent = emptyenv()),
+             envir = .rs.toolsEnv())
+
+   data <- chunkCallback
+   handler <- .Call("rs_createUUID")
+   assign(handler, value = data, envir = .rs.notebookChunkCallbacks)
+
+   return(handler)
+})
+
+# unregister a chunk callback functions
+.rs.addApiFunction("unregisterChunkCallback", function(handle) {
+  if (exists(".rs.notebookChunkCallbacks") &&
```

Comment:
I think we should emit a warning here if (a) the environment doesn't exist or (b) the handle wasn't found in the environment.